base/compiler/abstractinterpretation.jl

```diff
@@ -1363,6 +1365,13 @@ function matching_cache_argtypes(𝕃::AbstractLattice, mi::MethodInstance,
     else
         given_argtypes = va_process_argtypes(𝕃, given_argtypes, mi)
     end
+    if used !== nothing
+        for i = 1:length(given_argtypes)
+            if !used[i]
```

Comment:
Needs a comment why this doesn't break the concrete eval overwrite semantics.

Reply:
Probably this kind of widening should be done using `WidenedArgtypes`. But it would be fine as is, since unused-ness ensures that there's no contribution on rt and effects, etc. I basically wanted to propagate `Conditional` while allowing the argtype widening. I guess we probably need to refactor this `matching_cache_argtypes` interface more.